pkgs/development/python-modules/sphinxext-opengraph/default.nix

```diff
@@ -31,7 +31,7 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [
     sphinx
-    matplotlib
+    matplotlib # Optional dependency
```

Comment:
How about moving matplotlib to nativeCheckInputs and letting the user choose whether or not to install it?

Reply:
I did it in a different way.
Is that fine or are there any improvements that you can think of?

Reply:
I took inspiration from how `matrix-nio` handles optional dependencies to implement this.